youtube_dl/YoutubeDL.py

```diff
@@ -1291,6 +1289,22 @@ def process_info(self, info_dict):
             self.report_error('unable to create directory ' + compat_str(err))
             return
 
+        filesystem_enc = sys.getfilesystemencoding()
+        if filesystem_enc is None:
+            filesystem_enc = 'utf-8'
+        max_filename_len = 200
+        if sys.platform.startswith('linux'):
```

Comment:
I would prefer testing with `hasattr` instead to avoid [possible breakages](https://github.com/rg3/youtube-dl/commit/22603348aa0b3e02c520589dea092507a04ab06a) I've learned about the hard way.